src/Rings/mpoly-localizations.jl

```diff
@@ -3040,3 +3040,128 @@ end
   return dim(saturated_ideal(I))
 end
 
+################################################################################
+#
+# Minimal generating set
+#
+################################################################################
+
+@doc raw"""
+    minimal_generating_set(I::MPolyLocalizedIdeal)
+
+Given an ideal `I` in the localization of a multivariate polynomial ring
+over a field at a point, return an array containing a minimal set of
+generators of `I`. If `I` is the zero ideal an empty list is returned.
+
+# Examples
+```jldoctest
+julia> R, (x, y) = QQ["x", "y"];
+
+julia> L, phi = localization(R, complement_of_point_ideal(R, [1, 2]));
+
+julia> I = ideal(L, [x-1, y-2])^2
+Ideal
+  of localized ring
+with 4 generators
+  x^2 - 2*x + 1
+  x*y - 2*x - y + 2
+  x*y - 2*x - y + 2
+  y^2 - 4*y + 4
+
+julia> minimal_generating_set(I)
+3-element Vector{MPolyLocRingElem{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem, MPolyComplementOfKPointIdeal{QQField, QQFieldElem, QQMPolyRing, QQMPolyRingElem}}}:
+ y^2 - 4*y + 4
+ x*y - 2*x - y + 2
+ x^2 - 2*x + 1
+```
+"""
+@attr Vector{<:MPolyLocRingElem} function minimal_generating_set(
+    I::MPolyLocalizedIdeal{<:MPolyLocRing{<:Field, <:FieldElem,
+                                          <:MPolyRing, <:MPolyElem,
+                                          <:MPolyComplementOfKPointIdeal}
+                          }
+  )
+  L = base_ring(I)
+  R = base_ring(L)
+
+  @req coefficient_ring(R) isa AbstractAlgebra.Field "The coefficient ring must be a field"
+
+  shift, back_shift = base_ring_shifts(L)
+  I_shift = shifted_ideal(I)
+  o = negdegrevlex(R) # the default local ordering
+
+  if !isempty(I_shift.gb)
+
+    # use available gb for shifted ideal is available
+    G = first(values(I_shift.gb))
```

Comment:
Similar here.

Reply:
I agree that such an interface would be nice to have. But there is none. This code is just poking in the internals and might not even be safe. But for the moment, I'm afraid, we can't do better. It's a bigger task that we have to address here in the beginning of August.

Reply:
I put a guard in to make sure that a non-local ordering will not be used -- this could however throw an error. 
But I did not want to go through the whole list I_shift.gb to find a local one. I_shift is cached internal data for an ideal in a localized ring after all.